recipes/tbb/oneTBB/conanfile.py

```diff
@@ -0,0 +1,116 @@
+from conan.tools.cmake import CMake, CMakeToolchain
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import re
+
+required_conan_version = ">=1.37.0"
+
+class ConanFile(ConanFile):
```

Comment:
This actually emits warnings when using PyCharm:
```
Redeclared 'ConanFile' defined above without usage
```

Reply:
That is clearly a PyCharm bug, as the statement is false. The variable `ConanFile` is indeed redeclared, but the previous declaration been used, unlike what PyCharm is claiming -- the previous declaration of the variable `ConanFile` is used as the base class of the new class that is created and then assigned to the variable `ConanFile`. I would imagine PyCharm wouldn't complain if the usage of the `class` syntactic sugar was removed, e.g.
```python
ConanFile = type(
    "ConanFile",
    (ConanFile,),
    {
        "name": "tbb",
        "license": "Apache-2.0",
        ...
    }
)
```

This is functionally no different than other usages of shadow variables, e.g.
```python
foo_bar = " my string  "
foo_bar = foo_bar.strip()
```